---
title: "ChIP-Seq Quality report" 
author: "Author: David James (d.w.james@swansea.ac.uk)" 
date: "Last update: `r format(Sys.time(), '%d %B, %Y')`" 
output:
  BiocStyle::pdf_document: default
  BiocStyle::html_document:
    toc_float: true
    code_folding: show
package: systemPipeR
vignette: |
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{ChIP-Seq Workflow Template}
  %\VignetteEngine{knitr::rmarkdown}
#fontsize: 14pt
#bibliography: bibtex.bib
---

```{css, echo=FALSE}
pre code {
white-space: pre !important;
overflow-x: scroll !important;
word-break: keep-all !important;
word-wrap: initial !important;
}
---

# Preprocessing Quality
The following `seeFastq` and `seeFastqPlot` functions generate and plot a series of useful quality statistics for a set of FASTQ files including per cycle quality box
plots, base proportions, base-level quality trends, relative k-mer
diversity, length and occurrence distribution of reads, number of reads
above quality cutoffs and mean quality distribution. The results are
written to a PDF file named `fastqReport.pdf`.

```{r fastq_report, eval=TRUE}
#setwd("~/data/CurryExample/")
fqlist <- seeFastq(fastq=infile1(args), batchsize=10000, klength=8)
pdf("./results/fastqReport.pdf", height=18, width=4*length(fqlist))
seeFastqPlot(fqlist)
dev.off()

```